### PR TITLE
added reverse operation to reverse a patch

### PR DESCRIPTION
**Pull request by brandonpollack23**
I needed a simple undo operation for a project I'm considering working on.  I liked how simple diffy was to use so I added/tested this and thought maybe you would be so kind as to give feedback or accept it :)

### CODE REVIEW

**Comment by bmwill** [comment]
Thanks for the contribution! This looks good and fairly straightforward. I'll work on fixing the lint that is failing elsewhere in the code and then I'll get this merged. And sorry for the delay!

**Comment by brandonpollack23** [comment]
Oh yay thank you :)

File: src/diff/tests.rs [hunk 1]
```diff
@@ -664,3 +664,45 @@ Second:
...
+
+    let re_reverse = apply(&apply("", &p).unwrap(), &reverse).unwrap();
+    assert_eq!(re_reverse, "");
+}
+
+#[test]
+fn reverse_multi_line_file() {
+    let original = r"Commander Worf
+What do you want this time, Picard?!
+Commander Worf how dare you speak to mean that way!
```
Comment (correabuscar) [inline src/diff/tests.rs:694]: is `mean` here a typo? is it meant to `me` or `me in` ?

**Comment by brandonpollack23** [comment]
Me in would be correct.

**Pull request merged**